include/server/mir/scene/observer.h

```diff
@@ -33,18 +33,19 @@ class Surface;
 class Observer
 {
 public:
-    virtual void surface_added(Surface* surface) = 0;
-    virtual void surface_removed(Surface* surface) = 0;
+    virtual void surface_added(std::shared_ptr<Surface> surface) = 0;
```

Comment:
Please use `std::shared_ptr<Surface>const&` to avoid unnecessary copies and the barriers they need.

Reply:
Ack, will change when I get back to this

Reply:
Done